src/com/intellij/idea/plugin/hybris/runConfigurations/LocalSapCXRunSettingsEditor.kt

```diff
@@ -0,0 +1,147 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.intellij.idea.plugin.hybris.runConfigurations
+
+import com.intellij.compiler.options.CompileStepBeforeRun
+import com.intellij.diagnostic.logging.LogsGroupFragment
+import com.intellij.execution.ExecutionBundle
+import com.intellij.execution.JavaRunConfigurationExtensionManager
+import com.intellij.execution.ui.BeforeRunFragment
+import com.intellij.execution.ui.CommonParameterFragments
+import com.intellij.execution.ui.RunConfigurationFragmentedEditor
+import com.intellij.execution.ui.SettingsEditorFragment
+import com.intellij.idea.plugin.hybris.common.HybrisConstants
+import com.intellij.idea.plugin.hybris.common.utils.HybrisI18NBundleUtils
+import com.intellij.openapi.externalSystem.service.execution.configuration.addBeforeRunFragment
+import com.intellij.openapi.externalSystem.service.execution.configuration.addEnvironmentFragment
+import com.intellij.openapi.externalSystem.service.execution.configuration.fragments.SettingsEditorFragmentContainer
+import com.intellij.openapi.externalSystem.service.execution.configuration.fragments.addLabeledSettingsEditorFragment
+import com.intellij.openapi.externalSystem.service.ui.util.LabeledSettingsFragmentInfo
+import com.intellij.openapi.observable.properties.AtomicProperty
+import com.intellij.openapi.observable.util.bind
+import com.intellij.ui.components.fields.ExtendableTextField
+import com.intellij.ui.components.fields.IntegerField
+import org.jetbrains.idea.maven.execution.MavenRunConfiguration
+import org.jetbrains.idea.maven.project.MavenConfigurableBundle
+
+class LocalSapCXRunSettingsEditor(val runConfiguration: LocalSapCXRunConfiguration) :
+    RunConfigurationFragmentedEditor<LocalSapCXRunConfiguration>(runConfiguration, JavaRunConfigurationExtensionManager()) {
+
+    override fun createRunFragments(): List<SettingsEditorFragment<LocalSapCXRunConfiguration, *>> = SettingsEditorFragmentContainer.fragments {
+        add(CommonParameterFragments.createRunHeader())
+        addBeforeRunFragment(CompileStepBeforeRun.ID)
+        addAll(BeforeRunFragment.createGroup())
+        addDebugHostFragment()
+        addDebugPortFragment()
+        addDebugLineFragment(runConfiguration.getRemoteConnetion().launchCommandLine)
+        add(LogsGroupFragment())
+        addEnvironmentFragment()
+    }
+    private fun SettingsEditorFragmentContainer<LocalSapCXRunConfiguration>.addEnvironmentFragment() =
+        addEnvironmentFragment(
+            object : LabeledSettingsFragmentInfo {
+                override val editorLabel: String = ExecutionBundle.message("environment.variables.component.title")
+                override val settingsId: String = "maven.environment.variables.fragment"
```

Comment:
this part looks like copy paste from the `MavenRunConfigurationSettingsEditor`, let's make our own version.